adb/filesync_protocol.py

```diff
@@ -170,7 +170,7 @@ def Read(self, expected_ids, read_data=True):
 
     # Read one filesync packet off the recv buffer.
     header_data = self._ReadBuffered(self.recv_header_len)
-    header = struct.unpack(self.recv_header_format, header_data)
+    header = struct.unpack(self.recv_header_format, header_data.encode('ascii'))
```

Comment:
This again won't work for the same reason. `header_data` needs to be bytes when we get it since it's raw data.